tests/c/indirect_call.newcg.c

```diff
@@ -9,8 +9,6 @@
 //     jitstate: stop-tracing
 //     --- Begin jit-pre-opt ---
 //     ...
-//     %{{1}}: i32 = icall %{{2}}(%{{3}})
```

Comment:
@ltratt mem2reg optimise away the register which loads for `%3`.
For example: the entry block before mem2reg has `%3`, and `%11 ` loads it.
```
entry:
    %0: ptr = arg(0)
    %1: ptr = load_ti 40
    %2: ptr = load_ti 32
    %3: ptr = load_ti 24
    %4: ptr = load_ti 16
    %5: ptr = load_ti 8
    %6: ptr = load_ti 0
    tloop_start:
    %8: ptr = lookup_global @shadowstack_0
    *%8 = %6
    %10: ptr = load %2
    %11: i32 = load %3
    %12: ptr = ptr_add %6, 16
    %13: ptr = lookup_global @shadowstack_0
    *%13 = %12
    %15: i32 = icall %10(%11)
    %16: ptr = lookup_global @shadowstack_0
    *%16 = %6
```
However, when we enable `mem2reg` `%3` is removed so we don't have `%11` which was loading it and thus `icall` is optimised out.
```
entry:
   %0: ptr = arg(0)
   %1: ptr = load_ti 24
   %2: ptr = load_ti 16
   %3: ptr = load_ti 8
   %4: ptr = load_ti 0
   tloop_start:
   %6: ptr = lookup_global @shadowstack_0
   *%6 = %4
   %8: i32 = load %2
   %9: ptr = ptr_add %4, 16
   %10: ptr = lookup_global @shadowstack_0
   *%10 = %9
```

so seems like removing this line is right thing to do if we are turning on mem2reg?

Reply:
Huh, this looks like a bug to me, but the bug *might* be in the test.

Can you try adding `NOOPT_VAL(fn)` just before the `while` loop and see if the `icall` is still optimised away?

Reply:
@ltratt adding `NOOPT_VAL(fn)` fixes `icall` being optimised out. Thanks.